kernel/src/engine/mod.rs

```diff
@@ -11,7 +11,7 @@ pub(crate) mod arrow_conversion;
 ))]
 pub mod arrow_expression;
 #[cfg(feature = "arrow-expression")]
-pub(crate) mod arrow_utils;
+pub mod arrow_utils;
```

Comment:
I think we only want to make it `pub` if `internal_api` is set?

Reply:
absolutely.

Reply:
if this is just for a single function (pending `parse_json` comment above) should we just directly re-export it (under a feature flag) and not worry about making this whole module pub?

Reply:
hmm, good question. I guess so far we have been going pretty deep with our exports, but I personally like flatter structures a bit more. IMO in terms of developer convenience it only makes a real difference if we do a sweep through all of default engine?

Reply:
Yeah, for now I'd vote to just re-export under the feature flag. We can adjust if more gets exposed.